cmake/emu/simics.cmake

```diff
@@ -0,0 +1,30 @@
+# Copyright (c) 2023 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+
+find_program(
+  SIMICS
+  NAMES simics
+  )
+
+set(SIMICS_SCRIPT_PATH $ENV{SIMICS_SCRIPT_PATH})
+if(SIMICS_SCRIPT_PATH)
+  set(SIMICS_SCRIPT ${SIMICS_SCRIPT_PATH})
+else()
+  set(SIMICS_SCRIPT ${BOARD_DIR}/support/${BOARD}.simics)
+endif()
+
+get_property(SIMICS_ARGS GLOBAL PROPERTY "BOARD_EMU_ARGS_simics")
+
+add_custom_target(run_simics
```

Comment:
should this target be guarded to ensure it's only created if `SIMICS` was found ?

Optional: if simics was not found, then the `run_simics` target can be defined in a way that a useful message is printed to the user.

Reply:
sure,  this was following the pattern in all runners in the same folder.

Reply:
looked at all other emus, none is doing this check, not sure why this one should unless we have a common and unified pattern.

Reply:
no, but probably they should.

The reason for my comment is because if you try to execute `run_simics` and simics is not installed, then this is what you get:
```
$ ninja run_simics
...
[13/14] cd /projects/github/review/zephyr/build && SIMICS-NOTFOUND -no-gui -no-win /projects/github/review/zephyr/boards/intel/ish/support/intel_ish_5_8_0.simics project= zephyr_elf=/projects/github/review/zephyr/build/zephyr/zephyr.elf zephyr_start_address=0xff200000 -e run
/bin/sh: 1: SIMICS-NOTFOUND: not found
FAILED: zephyr/CMakeFiles/run_simics /projects/github/review/zephyr/build/zephyr/CMakeFiles/run_simics
```

which is not so helpful.
But it's also fine if we want to make such improvement across all runners as same time.
I just happened to notice it during review of this PR.

Reply:
right, same for `west build -b hifive1/fe310 samples/hello_world/ -t run` without renode installed....

```
[109/110] cd /home/nashif/zephyrproject/zephyr/build && RENODE-NOTFOUN...if/zephyrproject/zephyr/boards/sifive/hifive1/support/hifive1.resc; s'
/bin/sh: line 1: RENODE-NOTFOUND: command not found
FAILED: zephyr/CMakeFiles/run_renode /home/nashif/zephyrproject/zephyr/build/zephyr/CMakeFiles/run_renode
cd /home/nashif/zephyrproject/zephyr/build && RENODE-NOTFOUND --disable-xwt --port -2 --pid-file renode.pid -e '$elf=@/home/nashif/zephyrproject/zephyr/build/zephyr/zephyr.elf; include @/home/nashif/zephyrproject/zephyr/boards/sifive/hifive1/support/hifive1.resc; s'
ninja: build stopped: subcommand failed.
FATAL ERROR: command exited with status 1: /home/nashif/bin/cmake-3.21.1-linux-x86_64/bin/cmake --build /home/nashif/zephyrproject/zephyr/build --target run
```

I will followup and provide a fix for all emus